narwhals/_pandas_like/dataframe.py

```diff
@@ -106,6 +114,20 @@ def __getitem__(self, item: str | slice) -> PandasLikeSeries | PandasLikeDataFra
                 backend_version=self._backend_version,
             )
 
+        elif isinstance(item, tuple) and len(item) == 2:
+            from narwhals._pandas_like.series import PandasLikeSeries
+
+            if item[1] in self.columns:
+                native_series = self._native_dataframe.loc[item]
+            else:
+                native_series = self._native_dataframe.iloc[item]
```

Comment:
my understanding is that the the first item in the tuple is always a sequence of `int`s and the second item can be `str` or `int`. Given that we could have `int` column names in pandas, I thought that this `if item[1] in self.columns` should be ok to decide whether to use `iloc` or `loc`. 

but I feel like I am missing some edge case here 🤔 what do you think?

Reply:
I agree I didn't like myself what I committed. :D I really like this solution, thank you for adding `get_column`

FYI what I was trying to support is [this use case](https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/inspection/_permutation_importance.py#L67).
Here both rows and columns are accessed via position.